unstructured/partition/pdf.py

```diff
@@ -363,6 +364,38 @@ def _process_pdfminer_pages(
     return elements
 
 
+def convert_pdf_to_images(
+    filename: str = "",
+    file: Optional[Union[bytes, BinaryIO, SpooledTemporaryFile]] = None,
+    chunk_size: int = 10,
+) -> Iterator[List[object]]:
+    # Convert a PDF in small chunks of pages at a time (e.g. 1-10, 11-20... and so on)
+    exactly_one(filename=filename, file=file)
+    if file is not None:
+        f_bytes = convert_to_bytes(file)
+        info = pdf2image.pdfinfo_from_bytes(f_bytes)
+    else:
+        f_bytes = None
+        info = pdf2image.pdfinfo_from_path(filename)
+
+    total_pages = info["Pages"]
+    for start_page in range(1, total_pages + 1, chunk_size):
+        end_page = min(start_page + chunk_size - 1, total_pages)
+        if f_bytes is not None:
+            chunk_images = pdf2image.convert_from_bytes(
+                f_bytes,
+                first_page=start_page,
+                last_page=end_page,
+            )
+        else:
+            chunk_images = pdf2image.convert_from_path(
+                filename,
+                first_page=start_page,
+                last_page=end_page,
+            )
+        yield chunk_images
```

Comment:
The only improvement I can think of would be to `yield` the individual images rather than the chunks, while still handling things as chunks under the hood. However, I think what's here is good to commit rather than delaying. We can come back later and tweak the interface.

Reply:
@qued I got your point. At first, I also considered yielding individual images rather than chunks so I compared elapsed time when `chunk_size` is 1 and `chunk_size` is 10. The result showed that handling things as chunks is a bit faster than handling things as individual images. Please check the test result below. 

```
- chunk_size: 1
    - elapsed_time: 835.2164249420166
    - memory_peak: 255.3 MB
- chunk_size: 10
    - elapsed_time: 808.2655990123749
    - memory_peak: 277.3 MB
```

Reply:
To be clear, I'm saying we should load chunks, but yield individual images from the generator, so that we get the efficiency of chunks but the person using the generator doesn't have to manage the chunks.

Reply:
Sounds good! I'll get to that soon. Thanks for your suggestion! @qued